src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -963,7 +963,16 @@ void DynamoView_KeyDown(object sender, KeyEventArgs e)
 
         void DynamoView_KeyUp(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
+            if (e.Key == Key.Escape && dynamoViewModel.WatchEscapeIsDown)
+            {
+                dynamoViewModel.WatchEscapeIsDown = false;
+                dynamoViewModel.EscapeCommand.Execute(null);
+            }
+        }
+
+        void DynamoView_LostFocus(object sender, EventArgs e)
```

Comment:
Clicking on the search text box or a node will cause such an event as well if the ESC key is hold down. I think LostFocus will not take much time because it happens only once after the mouse is clicked. In comparison, when the ESC key is hold down, there will be many key down events.